.pipelines/tasks/iac-create-env.yml

```diff
@@ -0,0 +1,37 @@
+parameters:
+- name: ENVIRONMENT_NAME
+  displayName: Name of target environment
+  type: string
+
+jobs:
+- job: create_env_${{ parameters.ENVIRONMENT_NAME }}
+  dependsOn: IaC
+  variables:
+    # deviceName: ${{ join('edgeDevice',upper(parameters.ENVIRONMENT_NAME)) }}
+    deviceName: edgeDevice${{ parameters.ENVIRONMENT_NAME }}
+
+  steps:
+  - task: AzureCLI@2
+    displayName: 'Azure CLI: Create IoT Edge device for ${{ parameters.ENVIRONMENT_NAME }} environment'
+    inputs:
+      azureSubscription: $(AZURE_RESOURCE_MANAGER_CONNECTION_NAME)
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: >
+        az extension add --name azure-iot -y
+        && az iot hub device-identity create --hub-name $IOTHUB_NAME --device-id $(deviceName) --edge-enabled --output none
+        && echo "Created Edge device"
+        && sleep 5
+        && az iot hub device-twin update --device-id $(deviceName) --hub-name $IOTHUB_NAME --set tags=''{\"environment\":\"${{ parameters.ENVIRONMENT_NAME }}\"}'' && echo "Set tag for device" && CS_OUTPUT="$(az iot hub device-identity connection-string show --device-id $(deviceName) --hub-name $IOTHUB_NAME -o tsv)"
+        && echo "Got device connection string"
+        && echo "##vso[task.setvariable variable=CS_OUTPUT]${CS_OUTPUT}"
+
+  - task: AzureResourceGroupDeployment@2
+    displayName: 'Azure Deployment: Create VM for ${{ parameters.ENVIRONMENT_NAME }} environment'
+    inputs:
+      azureSubscription: $(AZURE_RESOURCE_MANAGER_CONNECTION_NAME)
+      resourceGroupName: '$(RG_NAME)'
+      location: '$(AZURE_LOCATION)'
+      templateLocation: 'URL of the file'
+      csmFileLink: 'https://raw.githubusercontent.com/Azure/iotedge-vm-deploy/1.2.0/edgeDeploy.json'
+      overrideParameters: '-deviceConnectionString $(CS_OUTPUT) -dnsLabelPrefix "$(IOTHUB_NAME)-${{ parameters.ENVIRONMENT_NAME }}" -adminUsername "devops" -authenticationType password -adminPasswordOrKey "$(VM_PASSWORD)" -vmSize "Standard_DS1_v2"'

```

Comment:
>-authenticationType password [](http://example.com/codeflow?start=156&length=28)

I got this error message below, I thing auth type is required to be ssh, but not sure.

2022-01-21T05:49:12.4803246Z There were errors in your deployment. Error code: InvalidTemplateDeployment.
2022-01-21T05:49:12.4806958Z ##[error]The template deployment failed because of policy violation. Please see details for more information.
2022-01-21T05:49:12.4828585Z ##[error]Details:
2022-01-21T05:49:12.4831608Z ##[error]Resource 'azureiotdevicesmarine-iothub-qa' was disallowed by policy. Error Type: PolicyViolation, Policy Definition Name : require ssh key for authentication on linux vms_1.9, Policy Assignment Name : SEC-DenyLinuxPwd-v1.9.
2022-01-21T05:49:12.4835479Z ##[error][More information on Azure Portal](https://portal.azure.com/#blade/Microsoft_Azure_Policy/EditAssignmentBlade/id/%252Fproviders%252FMicrosoft.Management%252FmanagementGroups%252F48fed3a1-0814-4847-88ce-b766155f2792%252Fproviders%252FMicrosoft.Authorization%252FpolicyAssignments%252FSEC-DenyLinuxPwd-v1.9)
2022-01-21T05:49:12.4839401Z ##[warning]Validation errors were found in the Azure Resource Manager template. This can potentially cause template deployment to fail. Task failed while creating or updating the template deployment.. Please follow [https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax)

Reply:
That's interesting. It works for me. Perhaps I don't have that policy in my subscription. Will try to come up with a workaround.

Reply:
Changed authentication type to ssh. VM_SSH_KEY variable should contain a public SSH Key for the device VM

Reply:
Generating an rsa key worked, thanks